libctru/source/services/fs.c

```diff
@@ -9,20 +9,30 @@
 #include <3ds/env.h>
 #include "../internal.h"
 
+#define FS_MAX_ARCHIVE_HANDLES 32
```

Comment:
I don't think that big of a handle table is needed, especially considering linear lookup is used (I was earlier suggesting 4). Otherwise good job.